src/lib/logical_query_plan/lqp_column_reference.cpp

```diff
@@ -26,10 +26,10 @@ std::ostream& operator<<(std::ostream& os, const LQPColumnReference& column_refe
   const auto original_node = column_reference.original_node();
   Assert(original_node, "OriginalNode has expired");
 
-  const auto stored_table_node = std::static_pointer_cast<const StoredTableNode>(column_reference.original_node());
+  const auto stored_table_node = std::dynamic_pointer_cast<const StoredTableNode>(column_reference.original_node());
```

Comment:
Why don't we need to check the result of the cast for a `nulltpr`?

Reply:
I was surprised to see only StoredTableNode here. For testing, I changed it to check for segfaults. In regular execution, we do not print MockNode expressions, but I fixed it anyway.